console/console-init/ui/src/modules/device/dialogs/CreateDevice/MetaDataRow.tsx

```diff
@@ -11,108 +11,165 @@ import {
+    let newRow = {
+      key: parentKey + "/",
+      value: [],
+      type: deviceRegistrationTypeOptions[0].value
+    };
+    setMetadataList([...metadataList, newRow]);
   };
 
   const handleDeleteRow = (index: any) => {
-    const metadataList = [...formData];
-    metadataList.splice(index, 1);
-    setFormData(metadataList);
+    const deletedRowMetadata = [...metadataList];
+    deletedRowMetadata.splice(index, 1);
+    setMetadataList(deletedRowMetadata);
   };
 
-  const handleInputChange = (e: any, index: any) => {
-    const { name, value } = e.target;
-    let metadataList = [...formData];
-    // metadataList[index][name]=value;
-    setFormData(metadataList);
+  const handleValueChange = (propertyValue: string, e: any) => {
+    let updatedValueMetadata = [...metadataList];
+    updatedValueMetadata[rowIndex].value = propertyValue;
+    setMetadataList(updatedValueMetadata);
   };
 
   const onPropertySelect = (e: any, selection: SelectOptionObject) => {
     setPropertySelected(selection.toString());
-    setPropertyInput(undefined);
   };
 
   const onPropertyClear = () => {
     setPropertySelected(undefined);
-    setPropertyInput(undefined);
   };
 
-  const typeOptions: ISelectOption[] = [
-    { key: "string", label: "String", value: "String" },
-    { key: "number", label: "Number", value: "Number" },
-    { key: "boolean", label: "Boolean", value: "Boolean" }
-  ];
+  //TODO: Clear value of property when type is array or object
+  let isChildAdditionEnabled = false;
+  if (metadataRow.type === "Object" || metadataRow.type === "Array") {
+    isChildAdditionEnabled = true;
+  }
 
+  //TODO: Increase width of type dropdown
   return (
     <>
       <Grid gutter="sm">
         <GridItem span={5}>
           <InputGroup>
-            <TypeAheadSelect
+            <Select
               id="cd-metadata-typeahead-parameter"
+              variant={SelectVariant.typeahead}
               ariaLabelTypeAhead={"Select parameter"}
-              ariaLabelledBy={"typeahead-parameter-id"}
-              onSelect={onPropertySelect}
+              onToggle={onToggle}
+              onSelect={onTypeAheadSelect}
               onClear={onPropertyClear}
-              selected={propertySelected}
-              inputData={propertyInput || ""}
+              selections={propertySelected || metadataRow.key}
+              isExpanded={isExpanded}
+              onFilter={onFilter}
+              ariaLabelledBy={"typeahead-parameter-id"}
               placeholderText={"Select property"}
-              onChangeInput={onChangePropertyInput}
-              setInput={setPropertyInput}
-            />
-            <Button
-              variant="control"
-              aria-label="Add child on button click"
-              onClick={e => handleAddChildRow(e, "context")}
-            >
-              <PlusIcon />
-            </Button>
+            ></Select>
+            {isChildAdditionEnabled === true ? (
```

Comment:
we can replace ternary operator by && like {isChildAdditionEnabled && (
<Button
                variant="control"
                aria-label="Add child on button click"
                onClick={e => {
                  let parentValue: string = metadataRow.key;
                  handleAddChildRow(e, parentValue);
                }}
              >
                <PlusIcon />
              </Button>
)}